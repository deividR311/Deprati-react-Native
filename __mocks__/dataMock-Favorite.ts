export const DataMock_EntryWishList = [
  {
    addedDate: '2022-08-17T23:51:15+0000',
    comment: '',
    desired: 1,
    priority: {
      code: 'medium',
      name: 'Wishlist2EntryPriority'
    },
    product: {
      acceptGiftPacking: false,
      availableForPickup: false,
      averageRating: 5,
      baseOptions: [
        {
          selected: {
            code: 'TRDEPRATI30',
            priceData: {
              currencyIso: 'USD',
              formattedValue: '$30,00',
              priceType: 'BUY',
              value: 30
            },
            stock: {
              stockLevelStatus: 'inStock'
            },
            url: '//p/TRDEPRATI30',
            variantOptionQualifiers: []
          },
          variantType: 'GiftCardVariantProduct'
        }
      ],
      baseProduct: 'TRDEPRATI',
      code: 'TRDEPRATI30',
      description:
        '<p style="margin-left:0cm; margin-right:0cm"><strong>Pol&iacute;ticas de uso:</strong>&nbsp;</p>\n\n<p style="margin-left:0cm; margin-right:0cm">La tarjeta regalo confiere al portador el derecho a:&nbsp;</p>\n\n<ul>\n\t<li>Comprar mercader&iacute;a &uacute;nicamente en Almacenes De Prati S.A. hasta por el valor pagado por la adquisici&oacute;n de la tarjeta.</li>\n\t<li>Exigir que conste en la nota de venta el valor consumido y el disponible.&nbsp;</li>\n\t<li>Consultar el saldo por cualquier de los medios que disponga&nbsp;Almacenes De Prati S.A.</li>\n</ul>\n\n<p style="margin-left:0cm; margin-right:0cm"><strong>Advertencia:&nbsp;</strong></p>\n\n<p style="margin-left:0cm; margin-right:0cm">Almacenes De Prati S.A. no se responsabiliza por sustracci&oacute;n o p&eacute;rdida de la Tarjeta Regalo cuando sea instrumento de compra del portador, ni de cualquier otra forma. En ning&uacute;n caso su valor ser&aacute; reembolsado en dinero.&nbsp;</p>\n',
      ean: 'JV30',
      images: [
        {
          altText: 'Tarjeta Regalo Para toda ocasión ',
          format: 'zoom',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h71/h00/8858638155806/JV30-0_product_515Wx772H'
        },
        {
          altText: 'Tarjeta Regalo Para toda ocasión ',
          format: 'product',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h31/h04/8858638286878/JV30-0_product_300Wx450H'
        },
        {
          altText: 'Tarjeta Regalo Para toda ocasión ',
          format: 'thumbnail',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h42/h07/8858638352414/JV30-0_product_96Wx144H'
        },
        {
          altText: 'Tarjeta Regalo Para toda ocasión ',
          format: 'cartIcon',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h40/h0a/8858638417950/JV30-0_product_65Wx97H'
        }
      ],
      installments: [],
      isGiftProduct: false,
      isPromotionSpecialPrice: false,
      ivaCondition: '0%',
      name: 'Tarjeta Regalo Para toda ocasión ',
      previousPrice: {},
      price: {
        currencyIso: 'USD',
        formattedValue: '$30,00',
        priceType: 'BUY',
        value: 30
      },
      purchasable: true,
      url: '//p/TRDEPRATI30'
    }
  },
  {
    addedDate: '2022-08-19T14:48:07+0000',
    comment: '',
    desired: 1,
    priority: {
      code: 'medium',
      name: 'Wishlist2EntryPriority'
    },
    product: {
      acceptGiftPacking: true,
      availableForPickup: false,
      baseOptions: [
        {
          selected: {
            code: '116040323800111009',
            priceData: {
              currencyIso: 'USD',
              formattedValue: '$26,78',
              priceType: 'BUY',
              value: 26.78
            },
            stock: {
              stockLevel: 15,
              stockLevelStatus: 'inStock'
            },
            url: '/jeans-slim-ho/p/116040323800111009',
            variantOptionQualifiers: []
          },
          variantType: 'GenericVariantProduct'
        }
      ],
      baseProduct: '000116040323800111',
      code: '116040323800111009',
      deliveryModes: [
        {
          code: 'standard'
        },
        {
          code: 'pickup'
        }
      ],
      ean: '15558873',
      images: [
        {
          format: 'zoom',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/hd3/h21/8900211441694/15558873-0_product_515Wx772H'
        },
        {
          format: 'product',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h13/h1e/8900211572766/15558873-0_product_300Wx450H'
        },
        {
          format: 'thumbnail',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h15/h1b/8900211638302/15558873-0_product_96Wx144H'
        },
        {
          format: 'cartIcon',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h04/h18/8900211703838/15558873-0_product_65Wx97H'
        }
      ],
      installments: [],
      isGiftProduct: false,
      isPromotionSpecialPrice: false,
      ivaCondition: '12%',
      name: 'Jeans Slim H&O',
      previousPrice: {},
      price: {
        currencyIso: 'USD',
        formattedValue: '$29,99',
        priceType: 'BUY',
        value: 29.9936
      },
      purchasable: true,
      url: '/jeans-slim-ho/p/116040323800111009'
    }
  },
  {
    addedDate: '2022-08-25T14:17:55+0000',
    comment: '',
    desired: 1,
    priority: {
      code: 'medium',
      name: 'Wishlist2EntryPriority'
    },
    product: {
      acceptGiftPacking: true,
      availableForPickup: false,
      averageRating: 3,
      baseOptions: [
        {
          selected: {
            code: '226A37851842157002',
            priceData: {
              currencyIso: 'USD',
              formattedValue: '$41,06',
              priceType: 'BUY',
              value: 41.06
            },
            stock: {
              stockLevel: 14,
              stockLevelStatus: 'inStock'
            },
            url: '/vestido-boho%2C-8%2C-ne/p/226A37851842157002',
            variantOptionQualifiers: []
          },
          variantType: 'GenericVariantProduct'
        }
      ],
      baseProduct: '226A37851842157',
      code: '226A37851842157002',
      deliveryModes: [
        {
          code: 'standard'
        },
        {
          code: 'pickup'
        }
      ],
      ean: '15444023',
      images: [
        {
          format: 'zoom',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/he7/h27/8888511365150/15444023-0_product_515Wx772H'
        },
        {
          format: 'product',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h28/h24/8888511496222/15444023-0_product_300Wx450H'
        },
        {
          format: 'thumbnail',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h17/h21/8888511561758/15444023-0_product_96Wx144H'
        },
        {
          format: 'cartIcon',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h19/h1e/8888511627294/15444023-0_product_65Wx97H'
        }
      ],
      installments: [],
      isGiftProduct: false,
      isPromotionSpecialPrice: false,
      ivaCondition: '12%',
      name: 'VESTIDO BOHO, 8, NE',
      previousPrice: {},
      price: {
        currencyIso: 'USD',
        formattedValue: '$45,99',
        priceType: 'BUY',
        value: 45.9872
      },
      purchasable: true,
      url: '/vestido-boho%2C-8%2C-ne/p/226A37851842157002'
    }
  },
  {
    addedDate: '2022-11-22T22:15:34+0000',
    comment: '',
    desired: 1,
    priority: {
      code: 'medium',
      name: 'Wishlist2EntryPriority'
    },
    product: {
      acceptGiftPacking: true,
      availableForPickup: false,
      averageRating: 4,
      baseOptions: [
        {
          selected: {
            code: '112010175904232001',
            priceData: {
              currencyIso: 'USD',
              formattedValue: '$2,23',
              priceType: 'BUY',
              value: 2.23
            },
            stock: {
              stockLevel: 6,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-naranja-con-bolsillo-aeropostale/p/112010175904232001',
            variantOptionQualifiers: []
          },
          variantType: 'GenericVariantProduct'
        }
      ],
      baseProduct: '000112010175904232',
      cloudflareVideo: {
        videoId: '0f7c1e29aac46b6d3d27ef77ecfcf5ad',
        width: '200',
        height: '200',
        responsive: true,
        autoplay: true,
        controls: false,
        loop: true,
        muted: false
      },
      code: '112010175904232001',
      deliveryModes: [
        {
          code: 'standard'
        },
        {
          code: 'pickup'
        }
      ],
      ean: '15543669',
      images: [
        {
          format: 'zoom',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h59/hc6/8894154342430/15543669-0_product_515Wx772H'
        },
        {
          format: 'product',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h9a/hc2/8894154473502/15543669-0_product_300Wx450H'
        },
        {
          format: 'thumbnail',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h9b/hbf/8894154539038/15543669-0_product_96Wx144H'
        },
        {
          format: 'cartIcon',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h8b/hbc/8894154604574/15543669-0_product_65Wx97H'
        }
      ],
      installments: [],
      isGiftProduct: false,
      isPromotionSpecialPrice: false,
      ivaCondition: '12%',
      name: 'Camiseta Aeropostale',
      previousPrice: {
        currencyIso: 'USD',
        formattedValue: '$24,99',
        priceType: 'BUY',
        value: 24.987199999999998
      },
      price: {
        currencyIso: 'USD',
        formattedValue: '$2,50',
        priceType: 'BUY',
        value: 2.4976
      },
      purchasable: true,
      url: '/camiseta-naranja-con-bolsillo-aeropostale/p/112010175904232001',
      youtubeMedia: {
        altText: 'Camiseta Aeropostale',
        videoID: 'o6L0KdG2S38',
        youtubeLink: 'https://youtu.be/o6L0KdG2S38',
        mediaFormat: 'youtubeVideo'
      }
    }
  },
  {
    addedDate: '2022-11-24T18:52:46+0000',
    comment: '',
    desired: 1,
    priority: {
      code: 'medium',
      name: 'Wishlist2EntryPriority'
    },
    product: {
      acceptGiftPacking: true,
      availableForPickup: false,
      baseOptions: [
        {
          selected: {
            code: '626A60202340002011',
            priceData: {
              currencyIso: 'USD',
              formattedValue: '$26,78',
              priceType: 'BUY',
              value: 26.78
            },
            stock: {
              stockLevel: 179,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-falda-baby-kiddo/p/626A60202340002011',
            variantOptionQualifiers: []
          },
          variantType: 'GenericVariantProduct'
        }
      ],
      baseProduct: '626A60202340002',
      code: '626A60202340002011',
      deliveryModes: [
        {
          code: 'standard'
        },
        {
          code: 'express'
        },
        {
          code: 'pickup'
        }
      ],
      ean: '15002252',
      images: [
        {
          format: 'zoom',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h2e/hcc/8869331828766/15002252-0_product_515Wx772H'
        },
        {
          format: 'product',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/hed/hcf/8869331959838/15002252-0_product_300Wx450H'
        },
        {
          format: 'thumbnail',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h55/he5/8869332025374/15002252-0_product_96Wx144H'
        },
        {
          format: 'cartIcon',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h94/he4/8869332090910/15002252-0_product_65Wx97H'
        }
      ],
      installments: [],
      isGiftProduct: false,
      isPromotionSpecialPrice: false,
      ivaCondition: '12%',
      name: 'Camiseta + falda Baby Kiddo',
      previousPrice: {},
      price: {
        currencyIso: 'USD',
        formattedValue: '$29,99',
        priceType: 'BUY',
        value: 29.9936
      },
      purchasable: true,
      url: '/camiseta-falda-baby-kiddo/p/626A60202340002011'
    }
  },
  {
    addedDate: '2023-01-27T20:34:19+0000',
    comment: '',
    desired: 1,
    priority: {
      code: 'medium',
      name: 'Wishlist2EntryPriority'
    },
    product: {
      acceptGiftPacking: true,
      availableForPickup: true,
      baseOptions: [
        {
          selected: {
            code: '223A00236312354003',
            priceData: {
              currencyIso: 'USD',
              formattedValue: '$32,13',
              priceType: 'BUY',
              value: 32.13
            },
            stock: {
              stockLevel: 0,
              stockLevelStatus: 'outOfStock'
            },
            url: '/blusa-sisa-alemis/p/223A00236312354003',
            variantOptionQualifiers: []
          },
          variantType: 'GenericVariantProduct'
        }
      ],
      baseProduct: '223A00236312354',
      code: '223A00236312354003',
      deliveryModes: [
        {
          code: 'standard'
        },
        {
          code: 'express'
        },
        {
          code: 'pickup'
        }
      ],
      ean: '16022252',
      images: [
        {
          format: 'zoom',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h0e/h29/8964489871390/16022252-0_product_515Wx772H'
        },
        {
          format: 'product',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h72/hd2/8964490002462/16022252-0_product_300Wx450H'
        },
        {
          format: 'thumbnail',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h20/hd3/8964490067998/16022252-0_product_96Wx144H'
        },
        {
          format: 'cartIcon',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h31/hd6/8964490133534/16022252-0_product_65Wx97H'
        }
      ],
      installments: [],
      isGiftProduct: false,
      isPromotionSpecialPrice: false,
      ivaCondition: '12%',
      name: 'Blusa Manga Sisa Alemis',
      previousPrice: {},
      price: {
        currencyIso: 'USD',
        formattedValue: '$35,99',
        priceType: 'BUY',
        value: 35.985600000000005
      },
      purchasable: true,
      url: '/blusa-sisa-alemis/p/223A00236312354003'
    }
  },
  {
    addedDate: '2023-01-30T20:46:57+0000',
    comment: '',
    desired: 1,
    priority: {
      code: 'medium',
      name: 'Wishlist2EntryPriority'
    },
    product: {
      acceptGiftPacking: false,
      availableForPickup: true,
      baseOptions: [],
      code: '739J60155009491',
      deliveryModes: [
        {
          code: 'standard'
        },
        {
          code: 'pickup'
        }
      ],
      description:
        '<p>El rubio perfecto, logra el tono que quieres sin causar ningun daño en tu cabello.</p>    <p>Detalles:</p>    <ul>   <li>Tinte Recamier Luminance</li>   <li>Tono: Rubio 7.0</li>   <li>Fórmula a base de keratina, miel, cera, colágeno y omega plus</li>   <li>Incluye:   <ul>    <li>Tratamiento pre color de 15 ml</li>    <li>Tubo de coloración en crema de 60 grs</li>    <li>Frasco revelador de color</li>    <li>Tubo de tratamiento post color de 60 grs</li>   </ul>   </li>   <li>Color brillante</li>   <li>Duración extra</li>   <li>Cubrimiento total de canas</li>  </ul>',
      ean: '12872875',
      images: [
        {
          altText: 'Tinte Luminance Rubio',
          format: 'zoom',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/hb2/hdd/8817446322206/12872875-0_product_515Wx772H'
        },
        {
          altText: 'Tinte Luminance Rubio',
          format: 'product',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/hf3/hd9/8817446453278/12872875-0_product_300Wx450H'
        },
        {
          altText: 'Tinte Luminance Rubio',
          format: 'thumbnail',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/hf6/hd6/8817446518814/12872875-0_product_96Wx144H'
        },
        {
          altText: 'Tinte Luminance Rubio',
          format: 'cartIcon',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h34/hd6/8817446584350/12872875-0_product_65Wx97H'
        }
      ],
      installments: [],
      isGiftProduct: false,
      isPromotionSpecialPrice: false,
      ivaCondition: '12%',
      name: 'Tinte Luminance Rubio',
      previousPrice: {},
      price: {
        currencyIso: 'USD',
        formattedValue: '$11,80',
        priceType: 'BUY',
        value: 11.8048
      },
      purchasable: true,
      url: '/tinte-luminance-rubio/p/739J60155009491'
    }
  },
  {
    addedDate: '2023-02-08T20:07:38+0000',
    comment: '',
    desired: 1,
    priority: {
      code: 'medium',
      name: 'Wishlist2EntryPriority'
    },
    product: {
      acceptGiftPacking: true,
      availableForPickup: false,
      baseOptions: [
        {
          selected: {
            code: '232A40311890319001',
            priceData: {
              currencyIso: 'USD',
              formattedValue: '$3,21',
              priceType: 'BUY',
              value: 3.21
            },
            stock: {
              stockLevel: 30,
              stockLevelStatus: 'inStock'
            },
            url: '/chompa-con-capucha%2C/p/232A40311890319001',
            variantOptionQualifiers: []
          },
          variantType: 'GenericVariantProduct'
        }
      ],
      baseProduct: '232A40311890319',
      code: '232A40311890319001',
      deliveryModes: [
        {
          code: 'standard'
        },
        {
          code: 'pickup'
        }
      ],
      ean: '15313362',
      images: [
        {
          format: 'zoom',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h0c/h4e/8891119206430/15313362-0_product_515Wx772H'
        },
        {
          format: 'product',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/hcb/h51/8891119337502/15313362-0_product_300Wx450H'
        },
        {
          format: 'thumbnail',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/hdc/h54/8891119403038/15313362-0_product_96Wx144H'
        },
        {
          format: 'cartIcon',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h8a/h55/8891119468574/15313362-0_product_65Wx97H'
        }
      ],
      installments: [],
      isGiftProduct: false,
      isPromotionSpecialPrice: false,
      ivaCondition: '12%',
      name: 'CHOMPA CON CAPUCHA,',
      previousPrice: {
        currencyIso: 'USD',
        formattedValue: '$35,99',
        priceType: 'BUY',
        value: 35.985600000000005
      },
      price: {
        currencyIso: 'USD',
        formattedValue: '$3,60',
        priceType: 'BUY',
        value: 3.5951999999999997
      },
      purchasable: true,
      url: '/chompa-con-capucha%2C/p/232A40311890319001'
    }
  },
  {
    addedDate: '2023-02-08T20:22:56+0000',
    comment: '',
    desired: 1,
    priority: {
      code: 'medium',
      name: 'Wishlist2EntryPriority'
    },
    product: {
      acceptGiftPacking: true,
      availableForPickup: false,
      baseOptions: [
        {
          selected: {
            code: '232030269405819001',
            priceData: {
              currencyIso: 'USD',
              formattedValue: '$2,50',
              priceType: 'BUY',
              value: 2.5
            },
            stock: {
              stockLevel: 21,
              stockLevelStatus: 'inStock'
            },
            url: '/capri-con-cortes%2C-x/p/232030269405819001',
            variantOptionQualifiers: []
          },
          variantType: 'GenericVariantProduct'
        }
      ],
      baseProduct: '000232030269405819',
      code: '232030269405819001',
      deliveryModes: [
        {
          code: 'standard'
        },
        {
          code: 'express'
        },
        {
          code: 'pickup'
        }
      ],
      ean: '15243596',
      images: [
        {
          format: 'zoom',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h83/h1f/8890309050398/15243596-0_product_515Wx772H'
        },
        {
          format: 'product',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/ha9/h00/8890309967902/15243596-0_product_300Wx450H'
        },
        {
          format: 'thumbnail',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/hd9/hf7/8890310033438/15243596-0_product_96Wx144H'
        },
        {
          format: 'cartIcon',
          imageType: 'PRIMARY',
          url: 'https://imagestg.deprati.com.ec/sys-master/images/h87/hf8/8890310098974/15243596-0_product_65Wx97H'
        }
      ],
      installments: [],
      isGiftProduct: false,
      isPromotionSpecialPrice: false,
      ivaCondition: '12%',
      name: 'CAPRI CON CORTES, X',
      previousPrice: {
        currencyIso: 'USD',
        formattedValue: '$27,99',
        priceType: 'BUY',
        value: 27.988799999999998
      },
      price: {
        currencyIso: 'USD',
        formattedValue: '$2,80',
        priceType: 'BUY',
        value: 2.8
      },
      purchasable: true,
      url: '/capri-con-cortes%2C-x/p/232030269405819001'
    }
  }
];
