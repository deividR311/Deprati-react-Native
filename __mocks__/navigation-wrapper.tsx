import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
