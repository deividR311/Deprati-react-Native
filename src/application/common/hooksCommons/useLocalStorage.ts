import { remove } from 'lodash';
import { useState, useEffect } from 'react';
import { saveStorage, getStorage } from '../../state-manager/storage/storage';

export const StatusEnum = {
  pending: 'pending',
  success: 'success',
  error: 'error'
};
type StatusEnumType = keyof typeof StatusEnum;

const useLocalStorage = (key, initialValue) => {
  const [status, setStatus] = useState<StatusEnumType>(StatusEnum.pending);
  // State to store our value
  // Pass initial state function to useState so logic is only executed once
  const [storedValue, setStoredValue] = useState(initialValue);

  useEffect(() => {
    getStorage(key, initialValue).then(storage => {
      setStoredValue(storage);
      setStatus(StatusEnum.success);
    });
  }, [key]);

  // Return a wrapped version of useState's setter function that ...
  // ... persists the new value to localStorage.
  const setValue = (value, keyUid?: string) => {
    try {
      // Allow value to be a function so we have same API as useState
      const valueToStore =
        value instanceof Function ? value(storedValue) : value;
      // Save state
      setStoredValue(valueToStore);
      // Save to local storage
      if (typeof saveStorage !== 'undefined') {
        saveStorage(keyUid ?? key, valueToStore);
      }
    } catch (error) {}
  };
  return [storedValue, setValue, status];
};

export default useLocalStorage;
