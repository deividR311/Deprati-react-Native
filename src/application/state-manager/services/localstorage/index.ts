export * from './interface';
export * from './localstorage.slice';
export * from './useLocalStorage';
