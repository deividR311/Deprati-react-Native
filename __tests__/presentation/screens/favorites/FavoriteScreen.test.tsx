import React from 'react';
import { render, waitFor, fireEvent, act } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import { useLocalStorage } from '../../../../src/application/state-manager/services/localstorage/useLocalStorage';
import useFavorite from '../../../../src/presentation/screens/favorite/Favorites/hook/useFavorite.hook';
import FavoriteScreen from '../../../../src/presentation/screens/favorite/Favorites/FavoritesScreen';
import { DataMock_EntryWishList } from '../../../../__mocks__/dataMock-Favorite';
import { IRender } from '../../../../__mocks__/IRender.interface';

jest.mock(
  '../../../../src/application/state-manager/services/localstorage/useLocalStorage'
);
jest.mock(
  '../../../../src/presentation/screens/favorite/Favorites/hook/useFavorite.hook'
);
let screenTest: IRender;
let isWitList: boolean = false;

describe('FavoriteScreen', () => {
  beforeEach(() => {
    // mockear useLocalStorage
    useLocalStorage.mockImplementation(() => ({
      save: jest.fn(),
      localStorageData: {
        IS_LOGIN: true // valor inicial de IS_LOGIN
      }
    }));
    // mockear useFavorite
    useFavorite.mockImplementation(() => {
      return {
        handleGetAllFavorites: jest.fn(),
        entryWishList: isWitList ? DataMock_EntryWishList : [],
        setProductSelect: (x: any) => jest.fn(x)
      };
    });
    screenTest = render(<NavigationWrapper screen={FavoriteScreen} />);
  });

  it('renders correctly FavoriteScreen in Snapshot', () => {
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });

  it('Empty Favorites', () => {
    const { getByTestId } = screenTest;
    expect(getByTestId('textEmpty')).toBeDefined();
    isWitList = true;
  });

  it('List With Favorites', () => {
    const { getByTestId } = screenTest;
    expect(getByTestId('listFavorites')).toBeDefined();
  });

  it('Item Favorite', async () => {
    const { getByTestId } = screenTest;
    expect(getByTestId('itemfavorite-6')).toBeDefined();
    expect(getByTestId('add-itemfavorite-6')).toBeDefined();
  });

  it('Item Favorite with size and color', async () => {
    const { getByTestId, getByText } = screenTest;

    expect(getByTestId('itemfavorite-0')).toBeDefined();
    expect(getByTestId('add-itemfavorite-0')).toBeDefined();

    const btnAdd = getByTestId('add-itemfavorite-0');
    fireEvent.press(btnAdd);
    expect(getByText('Confirma talla y color')).toBeDefined();
  });

  it('Item Favorite with size', async () => {
    const { getByTestId, getByText } = screenTest;
    expect(getByTestId('itemfavorite-1')).toBeDefined();
    expect(getByTestId('add-itemfavorite-1')).toBeDefined();

    const btnAdd = getByTestId('add-itemfavorite-1');
    fireEvent.press(btnAdd);

    expect(getByText('Confirma talla y color')).toBeDefined();

    const btnConfirm = getByText(/confirmar/i);
    expect(btnConfirm).toBeDefined();
  });

  it('Icon Shopping Cart', () => {
    const { getByTestId } = screenTest;
    expect(getByTestId('shopping-cart')).toBeDefined();
  });
});
