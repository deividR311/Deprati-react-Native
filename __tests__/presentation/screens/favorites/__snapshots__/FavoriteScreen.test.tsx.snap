// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteScreen renders correctly FavoriteScreen in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="Mi lista de deseos"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      >
        <RNSScreenStackHeaderSubview
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          type="right"
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginRight": 8,
                "opacity": 1,
              }
            }
            testID="shopping-cart"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 26,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              Ó£å
            </Text>
          </View>
        </RNSScreenStackHeaderSubview>
      </RNSScreenStackHeaderConfig>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {},
              Object {
                "backgroundColor": "#ffffff",
                "marginTop": -10,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "marginTop": 60,
                "paddingHorizontal": 12,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#263238",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 20,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                  "letterSpacing": 0.2,
                  "lineHeight": 24,
                  "textAlign": "center",
                }
              }
              testID="textEmpty"
            >
              Tu lista de favoritos est√° vac√≠a üôÅ, agrega productos a tus favoritos tocando el coraz√≥n ‚ù§.
            </Text>
            <View
              accessibilityLabel=""
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#E31A2C",
                  "borderRadius": 5,
                  "justifyContent": "center",
                  "marginTop": 30,
                  "opacity": 1,
                  "paddingVertical": 10,
                  "width": "95%",
                }
              }
              testID=""
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": undefined,
                      },
                      Object {},
                      Object {},
                    ]
                  }
                >
                  Regresar a inicio
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
